Replicator/Pusher.cc

```diff
@@ -227,6 +227,10 @@ namespace litecore { namespace repl {
             req[kConflictIncludesRevProperty] = "true"_sl;
         }
 
+        if (collectionIndex() != kNotCollectionIndex) {
```

Comment:
It should be Assert. If false, a lot of things won't work.

Reply:
I don't understand.  This property is not required to function for the BLIP message.  If omitted, it means "only the default collection".  What will collectionIndex be set to for legacy calls?

Reply:
I see, so what you are saying is not "this should be an assert" but rather "this check is not needed" ?